trade-finance-manager-api/src/v1/services/deal-cancellation/deal-cancellation.service.test.ts

```diff
@@ -0,0 +1,118 @@
+import { TfmDealCancellation } from '@ukef/dtfs2-common';
+import { CANCEL_DEAL_FUTURE_DATE, CANCEL_DEAL_PAST_DATE } from '../../../constants/email-template-ids';
+import sendTfmEmail from '../send-tfm-email';
+import { DealCancellationService } from './deal-cancellation.service';
+
+const mockPimEmailAddress = 'pim@example.com';
+const mockFormattedFacilities = 'mock formatted facilities';
+const ukefDealId = 'ukefDealId';
+const ukefFacilityIds = ['aFacilityId'];
+
+jest.mock('../send-tfm-email');
+jest.mock('../../api', () => ({
+  findOneTeam: jest.fn(() => ({ email: mockPimEmailAddress })),
+  findOneDeal: jest.fn(() => ({ dealSnapshot: { ukefDealId } })),
+  findFacilitiesByDealId: jest.fn(() =>
+    ukefFacilityIds.map((ukefFacilityId) => ({
+      facilitySnapshot: {
+        ukefFacilityId,
+      },
+    })),
+  ),
+}));
+
+jest.mock('./helpers/format-facility-ids', () => ({
+  formatFacilityIds: jest.fn(() => mockFormattedFacilities),
+}));
+
+const dealId = 'dealId';
+
+describe('sendDealCancellationEmail', () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(1729089801730);
```

Comment:
Is this really necessary, could we use `new Date` or similar instead?

Reply:
This allows me to hard code the output as `'16 October 2024'` without having to use date-fns to format it.
I have a slight preference towards this, since then the tests arent assuming i've written `dd MMMM yyyy` correctly

Reply:
Can we use date-fns? I'm finding these mysterious; Currently if a future developer was to come into this file, the hard coded dates don't have any meaning or context with these:

```ts
setSystemTime(1729089801730);

bankRequestDate: 1729089801730,
effectiveFrom: 1729176201730
```

I.e I have no idea what these dates actually are.

Reply:
Sure, I see that & its maybe more clear to use some constants at the top. I've switched to using `format` and `add` from date-fns though